Assets/MixedRealityToolkit-SDK/Features/Input/Handlers/ManipulationHandler.cs

```diff
@@ -0,0 +1,489 @@
+                                                                            Time.unscaledDeltaTime.ToString();
+                }
+
+                GameObject.Find("DebugText").GetComponent<TextMesh>().text = eventData.SourceId.ToString();
+            }
+
+            Vector3 inputPosition = Vector3.zero;
+            m_handsPressedInputSourceMap[eventData.SourceId] = eventData.InputSource;
+
+            // Add to hand map
+            if (eventData.InputSource.SourceName.Contains("Hand") == true)
+            {
+                inputPosition = m_handsPoseMap[eventData.SourceId];
+            }
+            else
+            {
+                eventData.InputSource.Pointers[0].TryGetPointerPosition(out inputPosition);   
+            }
+            m_handsPressedLocationsMap[eventData.SourceId] = inputPosition;
+
+            UpdateStateMachine();
+            eventData.Use();
+        }
+
+        /// <summary>
+        /// Event Handler receives input from inputSource
+        /// </summary>
+        public void OnInputUp(InputEventData eventData)
+        {
+            RemoveSourceIdFromHandMap(eventData.SourceId);
+            UpdateStateMachine();
+            eventData.Use();
+        }
+
+        /// <summary>
+        /// from IMixedRealitySpacialInputHandler
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnPoseInputChanged(InputEventData<MixedRealityPose> eventData)
+        {
+            if (debugText)
+            {
+           //    debugText.text = "On pose changed " + eventData.MixedRealityInputAction.Description + " " + Time.unscaledDeltaTime.ToString();
+            }
+            if (eventData.InputSource.SourceName.Contains("Hand"))
+            {
+                m_handsPoseMap[eventData.SourceId] = eventData.InputData.Position;
+            }
+        }
+
+        /// <summary>
+        /// OnSourceLost
+        /// </summary>
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            RemoveSourceIdFromHandMap(eventData.SourceId);
+            UpdateStateMachine();
+            eventData.Use();
+        }
+        #endregion Event Handlers
+
+        #region Unused Event Handlers
+        /// <summary>
+        /// OnSourceDetected Event Handler
+        /// </summary>
+        public void OnSourceDetected(SourceStateEventData eventData) { }
+        public void OnInputPressed(InputEventData<float> eventData)
+        {
+        }
+
+        public void OnPositionInputChanged(InputEventData<Vector2> eventData)
+        {
+        }
+
+        public void OnPositionChanged(InputEventData<Vector3> eventData)
+        {
+        }
+
+        public void OnRotationChanged(InputEventData<Quaternion> eventData)
+        {
+        }
+        #endregion Unused Event Handlers
+
+        #region Private Event Handlers
+        private void OnTwoHandManipulationUpdated()
+        {
+            
+#if UNITY_2017_2_OR_NEWER
```

Comment:
MRTK vNext requires Unity 2018, it is likely safe to remove this precompiler condition.

same comment applies to all UNITY_2017_x_OR_NEWER conditions